core/storage/question/gae_models_test.py

```diff
@@ -388,6 +388,56 @@ def test_get_question_skill_links_based_on_difficulty(self):
         self.assertTrue(questionskilllink_model2 in question_skill_links)
         self.assertTrue(questionskilllink_model4 in question_skill_links)
 
+    def test_get_random_question_skill_links_based_on_difficulty(self):
+        questionskilllink_model1 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id1', 'skill_id1', 0.3)
+            )
+        questionskilllink_model2 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id2', 'skill_id1', 0.6)
+            )
+        questionskilllink_model3 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id3', 'skill_id1', 0.3)
+            )
+        question_models.QuestionSkillLinkModel.put_multi_question_skill_links(
+            [questionskilllink_model1, questionskilllink_model2,
+             questionskilllink_model3])
+        question_skill_links = (
+            question_models.QuestionSkillLinkModel.
+            get_question_skill_links_based_on_difficulty_equidistributed_by_skill( # pylint: disable=line-too-long
+                2, ['skill_id1'], 0.6
+            )
+        )
+        self.assertEqual(len(question_skill_links), 2)
+        self.assertTrue(questionskilllink_model2 in question_skill_links)
+        self.assertTrue(questionskilllink_model1 in question_skill_links or
```

Comment:
By testing this way you aren't really checking if the randomization is working as expected. Here's what you can do - set a random seed, call the function, verify the output. Change the random seed, call the function, verify that the output is now different

Reply:
Done.